pkg/server/stream/v3/subscription.go

```diff
@@ -39,8 +48,8 @@ func newSubscription(wildcard bool, initialResourceVersions map[string]string) S
 	return state
 }
 
-func NewSotwSubscription(subscribed []string) Subscription {
-	sub := newSubscription(len(subscribed) == 0, nil)
+func NewSotwSubscription(subscribed []string, opts config.Opts, typeURL string) Subscription {
```

Comment:
Why is this passed in this interface? This is still leaking parameters here that have no value on the subscription